performance-tuning-methods.md

```diff
@@ -0,0 +1,281 @@
+
+- Database Time By SQL Type
+- Database Time By SQL Phase
+- SQL Execute Time Overview
+
+#### Tune by color
+
+The diagrams of database time breakdown and execution time overview present both expected and unexpected time consumption intuitively. Therefore, you can quickly locate performance bottleneck and learn the load profile. Green and blue areas stand for normal time consumption and requests. If non-green or non-blue areas occupy a significant proportion in these two diagrams, the database time distribution is inappropriate.
+
+- Database Time By SQL Type:
+
+    - Blue: `Select` statement
+    - Green: `Update`, `Insert`, `Commit` and other DML statements
+    - Red: General SQL types, including `StmtPrepare`, `StmtReset`, `StmtFetch`, and `StmtClose`
+
+- Database Time By SQL Phase: The SQL execution phase is in green and other phases are in red on general. If non-green areas are large, it means much database time is consumed in other phases than the execution phase and further cause analysis is required. A common scenario is that the compile phase shown in orange takes a large area due to unavailability of the execution plan cache feature.
+- SQL Execute Time Overview: Green metrics stand for common KV write requests (such as Prewrite and Commit), blue metrics stand for common KV read requests (such as Cop and Get), and metrics in other colors stand for unexpected situations which you need to pay attention. For example, pessimistic lock KV requests are marked red and TSO waiting is marked dark brown. If non-blue or non-green areas are large, it means there is bottleneck during SQL execution. For example:
+
+    - If serious lock conflicts occur, the red area will take a large proportion.
+    - If excessive time is consumed in waiting TSO, the dark brown area will take a large proportion.
+
+**Example 1: TPC-C load**
+
+![TPC-C](/media/performance/tpcc_db_time.png)
+
+- Database Time by SQL Type: Most time-consuming statements are commit, update, select and insert statements.
+- Database Time by SQL Phase: The most time-consuming phase is SQL execution in green.
+- SQL Execute Time Overview: The most time-consuming KV requests in SQL execution are Prewrite and Commit in green.
+
+    > **Note:**
+    >
+    > It is normal that the total KV request time is greater than the execute time. Because the TiDB executor may send KV requests to multiple TiKVs concurrently, causing the total KV request wait time to be greater than the execute time. In the preceding TPC-C load, TiDB sends Prewrite and Commit requests concurrently to multiple TiKVs when a transaction is committed. Therefore, the total time for Prewrite, Commit and PessimisticsLock requests in this example is obviously longer than the execute time.
```

Comment:
```suggestion
    > It is normal that the total KV request time is greater than the `execute` time. Because the TiDB executor may send KV requests to multiple TiKVs concurrently, causing the total KV request wait time to be greater than the `execute` time. In the preceding TPC-C load, TiDB sends `Prewrite` and `Commit` requests concurrently to multiple TiKVs when a transaction is committed. Therefore, the total time for `Prewrite`, `Commit` and `PessimisticsLock` requests in this example is obviously longer than the `execute` time.
```